src/literal.h

```diff
@@ -33,25 +33,27 @@ class Literals;
 struct ExceptionPackage;
 
 class Literal {
+
+private:
   // store only integers, whose bits are deterministic. floats
   // can have their signalling bit set, for example.
   union {
     int32_t i32;
     int64_t i64;
     uint8_t v128[16];
-    Name func; // function name for funcref
+    // funcref function name. not set indicates `null`.
```

Comment:
Instead of `not set` (which sounds dangerous), how about specifying "empty string" and "null pointer"